app/interactors/convert_exams.rb

```diff
@@ -0,0 +1,61 @@
+require 'active_support/core_ext/numeric/time'
+require 'interpipe/interactor'
+
+class ConvertExams
+  include Interpipe::Interactor
+
+  DEFAULT_EXAM_DURATION = 90.minutes
+
+  def perform(exams:, faculty_semester:, rooms:, **args)
+    @courses = {}
+    @people = {}
+    @faculty_semester = faculty_semester
+    @rooms_map = build_room_map(rooms)
+    @events = exams.map do |exam|
+      convert_exam(exam)
+    end
+  end
+
+  def results
+    {
+      events: @events,
+      courses: @courses.values,
+      people: @people.values
+    }
+  end
+
+  def convert_exam(exam)
+    event = Event.new
+    event.starts_at = exam.start_date
+    event.ends_at = (exam.end_date ? exam.end_date : exam.start_date + DEFAULT_EXAM_DURATION)
```

Comment:
I've created special role `ImportedExam` for dealing with some of the import logic and refactored the method under question.